### PR TITLE
[ux.symfony.com] Add demo page for Translator

### PR DESCRIPTION
**Pull request by Kocal**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no 
| Tickets       | Fix #... 
| License       | MIT

Needs #822 

Added interactive demo page for the Translator package, where the user can see the code, the output, and some inputs to play with.

https://user-images.githubusercontent.com/2103975/235448398-5cc4e45f-8d5f-475f-8351-c43d57f6a3c7.mov

### CODE REVIEW

File: src/Translator/assets/dist/translator.d.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ export type TranslationsType = Record<DomainType, {
     parameters: ParametersType;
 }>;
 export type NoParametersType = Record<string, never>;
-export type ParametersType = Record<string, string | number> | NoParametersType;
+export type ParametersType = Record<string, string | number | Date> | NoParametersType;
```
Comment (Kocal) [inline src/Translator/assets/dist/translator.d.ts:7]: This is fix is cherrypicked in #822

File: src/Translator/assets/src/translator.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ export type LocaleType = string;
 
 export type TranslationsType = Record<DomainType, { parameters: ParametersType }>;
 export type NoParametersType = Record<string, never>;
-export type ParametersType = Record<string, string | number> | NoParametersType;
+export type ParametersType = Record<string, string | number | Date> | NoParametersType;
```
Comment (Kocal) [inline src/Translator/assets/src/translator.ts]: Also fixed in #822

File: src/Translator/doc/index.rst [hunk 3]
```diff
@@ -61,16 +61,17 @@ Configuring the default locale
...
+#. Or with ``<html lang="your-locale">`` attribute
 
 Importing and using translations
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-If you use the Symfony Flex recipe, you can import the ``trans()`` function and your translations in your assets from the file `assets/translator.js`.
+If you use the Symfony Flex recipe, you can import the ``trans()`` function and your translations in your assets from the file ``assets/translator.js``.
 
-Translations are available as named exports, by using the translation's id transformed in uppercase snake-case (e.g.: `my.translation` becomes `MY_TRANSLATION`),
+Translations are available as named exports, by using the translation's id transformed in uppercase snake-case (e.g.: ``my.translation`` becomes ``MY_TRANSLATION``),
```
Comment (Kocal) [inline src/Translator/doc/index.rst]: Also fixed in #822

File: ux.symfony.com/templates/ux_packages/translator.html.twig [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+        eyebrowText: 'eyebrow text'
+    } %}
+        {% block title_header %}
+            title
+            <span class="playfair ps-2">header</span>
+        {% endblock %}
+
+        {% block sub_content %}
+            sub_content
+        {% endblock %}
```
Comment (Kocal) [inline ux.symfony.com/templates/ux_packages/translator.html.twig]: I need help for wordings in eyebrow, title header and sub content texts, I'm super bad at it :s

File: ux.symfony.com/assets/styles/_code.scss [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  padding: 12px;
+  border-radius: 6px;
+  overflow-x: auto;
+
+  background-color: $n-800;
+  height: 100%;
+  font-family: 'Space Mono', monospace;
+  font-size: 14px;
+  line-height: 160%;
+  color: #FFF;
```
Comment (weaverryan) [inline ux.symfony.com/assets/styles/_code.scss]: These bottom styles are the same as `.termina-body`, right? We should just use that (yea, I know at this point the class name isn't ideal as we use it for code).

File: ux.symfony.com/templates/ux_packages/translator.html.twig [hunk 6]
```diff
@@ -0,0 +1,251 @@
+{% extends 'packageBase.html.twig' %}
+
+{% block component_header %}
+    {% component PackageHeader with {
+        package: 'translator',
+        eyebrowText: 'eyebrow text'
+    } %}
+        {% block title_header %}
+            title
+            <span class="playfair ps-2">header</span>
```
Comment (weaverryan) [inline ux.symfony.com/templates/ux_packages/translator.html.twig]: Translations directly imported in *JavaScript*

File: ux.symfony.com/templates/ux_packages/translator.html.twig [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+        {% endblock %}
+
+        {% block sub_content %}
+            sub_content
```
Comment (weaverryan) [inline ux.symfony.com/templates/ux_packages/translator.html.twig]: Write & manage your translations like you always do. Then use them just as easily in JS.

File: ux.symfony.com/templates/ux_packages/translator.html.twig [hunk 8]
```diff
@@ -0,0 +1,251 @@
+{% extends 'packageBase.html.twig' %}
+
+{% block component_header %}
+    {% component PackageHeader with {
+        package: 'translator',
+        eyebrowText: 'eyebrow text'
```
Comment (weaverryan) [inline ux.symfony.com/templates/ux_packages/translator.html.twig]: JavaScript-enabled Translation

**Review by weaverryan — COMMENTED** [review]
Looks great! I suggested some text - it's especially hard to fill 3 spaces of text without sounding repetitive.

The one thing I feel like we're missing in the demo is the `import`. I see the translation files on top - e.g. `say_hello` - then suddenly `trans(SAY_HELLO)` works. Not seeing the import gives me trouble seeing how this works.

Easiest solution is to add the import to the top of every code bock:

```
import { SAY_HELLO, trans } from './translator';
```

That would definitely solve the problem - but it could look "busy" since it will be 1 extra line added to ALL of the code blocks.

File: ux.symfony.com/templates/ux_packages/translator.html.twig (hunk 6 above)
Comment (Kocal) [reply ux.symfony.com/templates/ux_packages/translator.html.twig]: Thanks!

File: ux.symfony.com/assets/styles/_code.scss (hunk 5 above)
Comment (Kocal) [reply ux.symfony.com/assets/styles/_code.scss]: Fixed

**Comment by weaverryan** [comment]
Thank you @Kocal!

**Pull request merged**